packages/bottom-tabs/src/views/BottomTabBar.tsx

```diff
@@ -249,18 +260,87 @@ export default function BottomTabBar({
     descriptors,
     dimensions,
     layout,
+    tabBarPosition,
   });
 
   const tabBarBackgroundElement = tabBarBackground?.();
 
+  const tabBarPositionMap: TabBarPositionMapConfig = {
+    right: {
+      tabBarItem: {
+        height:
+          Platform.OS === ('ios' || 'android')
+            ? tabBarHeight / 1.5
+            : DEFAULT_TABBAR_HEIGHT,
+        marginTop: COMPACT_TABBAR_HEIGHT / 2,
+        flex: Platform.OS === ('ios' || 'android') ? 0 : 0.07,
+      },
+      tabBar: {
+        paddingHorizontal: 5,
+        borderLeftColor: colors.border,
+        paddingTop: insets.top + tabBarHeight,
+        borderLeftWidth: StyleSheet.hairlineWidth,
+      },
+      content: {
+        flex: 1,
+        width: tabBarHeight,
+        maxWidth:
+          Platform.OS === ('ios' || 'android')
+            ? DEFAULT_TABBAR_HEIGHT
+            : DEFAULT_TABBAR_HEIGHT * 1.3,
+      },
+    },
+
+    left: {
+      tabBarItem: {
+        height:
+          Platform.OS === ('ios' || 'android')
+            ? tabBarHeight / 1.5
+            : DEFAULT_TABBAR_HEIGHT,
+        marginTop: COMPACT_TABBAR_HEIGHT / 2,
+        flex: Platform.OS === ('ios' || 'android') ? 0 : 0.07,
+      },
+      tabBar: {
+        paddingHorizontal: 5,
+        borderRightColor: colors.border,
+        paddingTop: insets.top + tabBarHeight,
+        borderRightWidth: StyleSheet.hairlineWidth,
+      },
+      content: {
+        flex: 1,
+        width: tabBarHeight,
+        maxWidth:
+          Platform.OS === ('ios' || 'android')
+            ? DEFAULT_TABBAR_HEIGHT
+            : DEFAULT_TABBAR_HEIGHT * 1.3,
+      },
+    },
+
+    bottom: {
+      tabBarItem: {},
+      tabBar: {
+        height: tabBarHeight,
+        borderTopColor: colors.border,
+        borderTopWidth: StyleSheet.hairlineWidth,
+        paddingHorizontal: Math.max(insets.left, insets.right),
+      },
+      content: {
+        flex: 1,
+        flexDirection: 'row',
+      },
+    },
+  };
+
   return (
     <Animated.View
       style={[
-        styles.tabBar,
+        tabBarPositionMap[tabBarPosition].tabBar,
         {
+          left: 0,
+          right: 0,
+          bottom: 0,
```

Comment:
The code conventions in this repo are to keep static styles in `StyleSheet` and only define dynamic styles inside the component. Let's keep that convention.

Reply:
got you